service_test.go

```diff
@@ -277,6 +290,77 @@ func TestHandleEvent(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, NIP_47_ERROR_RESTRICTED, received.Error.Code)
 	assert.NotNil(t, res)
+
+	// pay_keysend: without permission
+	newPayload, err = nip04.Encrypt(nip47KeysendJson, ss)
+	assert.NoError(t, err)
+	res, err = svc.HandleEvent(ctx, &nostr.Event{
+		ID:      "test_pay_keysend_event_1",
+		Kind:    NIP_47_REQUEST_KIND,
+		PubKey:  senderPubkey,
+		Content: newPayload,
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, res)
+	decrypted, err = nip04.Decrypt(res.Content, ss)
+	assert.NoError(t, err)
+	received = &Nip47Response{}
+	err = json.Unmarshal([]byte(decrypted), received)
+	assert.NoError(t, err)
+	assert.Equal(t, NIP_47_ERROR_RESTRICTED, received.Error.Code)
+	assert.NotNil(t, res)
+	// pay_keysend: with permission
+	// update the existing permission to pay_invoice so we can have the budget info and increase max amount
+	newMaxAmount = 1000
+	err = svc.db.Model(&AppPermission{}).Where("app_id = ?", app.ID).Update("request_method", NIP_47_PAY_INVOICE_METHOD).Update("max_amount", newMaxAmount).Error
+	assert.NoError(t, err)
+	// create a second permission for getting the budget
+	appPermission = &AppPermission{
+		AppId:         app.ID,
+		App:           app,
+		RequestMethod: NIP_47_PAY_KEYSEND_METHOD,
+		ExpiresAt:     expiresAt,
+	}
+	err = svc.db.Create(appPermission).Error
+	res, err = svc.HandleEvent(ctx, &nostr.Event{
+		ID:      "test_pay_keysend_event_2",
+		Kind:    NIP_47_REQUEST_KIND,
+		PubKey:  senderPubkey,
+		Content: newPayload,
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, res)
+	decrypted, err = nip04.Decrypt(res.Content, ss)
+	assert.NoError(t, err)
+	received = &Nip47Response{
+		Result: &Nip47PayResponse{},
+	}
+	err = json.Unmarshal([]byte(decrypted), received)
+	assert.NoError(t, err)
+	assert.Equal(t, "123preimage", received.Result.(*Nip47PayResponse).Preimage)
+
+	// keysend: budget overflow
+	newMaxAmount = 100
+	// we change the budget info in pay_invoice permission
+	err = svc.db.Model(&AppPermission{}).Where("request_method = ?", NIP_47_PAY_INVOICE_METHOD).Update("max_amount", newMaxAmount).Error
```

Comment:
this is using the wrong NIP-47 method

Reply:
[Discussion](https://getalby.slack.com/archives/C04UJ0USX70/p1702456752694139) on slack